libgnucash/engine/Account.cpp

```diff
@@ -3259,35 +3262,18 @@ gnc_account_get_full_name(const Account *account)
     /* errors */
     g_return_val_if_fail(GNC_IS_ACCOUNT(account), g_strdup(""));
 
-    /* optimizations */
-    priv = GET_PRIVATE(account);
-    if (!priv->parent)
-        return g_strdup("");
-
-    /* Figure out how much space is needed by counting the nodes up to
-     * the root. */
-    level = 0;
-    for (a = account; a; a = priv->parent)
-    {
-        priv = GET_PRIVATE(a);
-        level++;
-    }
-
-    /* Get all the pointers in the right order. The root node "entry"
-     * becomes the terminating nullptr pointer for the array of strings. */
-    names = (const gchar **)g_malloc(level * sizeof(gchar *));
-    names[--level] = nullptr;
-    for (a = account; level > 0; a = priv->parent)
-    {
-        priv = GET_PRIVATE(a);
-        names[--level] = priv->accountName;
-    }
+    auto path{gnc_account_get_path (account)};
+    auto gs = g_string_new (nullptr);
```

Comment:
This is severely pessimized over what you're replacing because of the repeated allocations. Call [`g_string_sized_new`](https://docs.gtk.org/glib/ctor.String.sized_new.html) to do the allocation in one go. You could get the size with another loop or just allocate a big buffer, maybe 512 or 1024 bytes.